src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -166,42 +159,33 @@ public override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage req
         {
             Debug.Assert(async);
 
-            // Wait for an available stream (based on QUIC MAX_STREAMS) if there isn't one available yet.
-
-            TaskCompletionSourceWithCancellation<bool>? waitForAvailableStreamTcs = null;
-
-            lock (SyncObj)
-            {
-                long remaining = _requestStreamsRemaining;
-
-                if (remaining > 0)
-                {
-                    _requestStreamsRemaining = remaining - 1;
-                }
-                else
-                {
-                    waitForAvailableStreamTcs = new TaskCompletionSourceWithCancellation<bool>();
-                    _waitingRequests.Enqueue(waitForAvailableStreamTcs);
-                }
-            }
-
-            if (waitForAvailableStreamTcs != null)
-            {
-                await waitForAvailableStreamTcs.WaitWithCancellationAsync(cancellationToken).ConfigureAwait(false);
-            }
-
             // Allocate an active request
-
             QuicStream? quicStream = null;
             Http3RequestStream? requestStream = null;
 
             try
             {
+                if (_connection != null)
```

Comment:
Should we possibly fail fast from the start, if `_connection` is null? We wouldn't wait to acquire lock around `_activeRequests` down there (line 185) that we don't actually need in this case, and it will also reduce block nesting depth (or whatever it's called).

Reply:
What I have here is actually wrong, we should check the `_connection` inside the lock. We handle all the changes to it inside a lock. So I have a potential race here. I need to fix this.
Thanks for pointing at this code and the locking!

Reply:
I also reorder the statements here, so your point is addressed.